ballerina/utils.bal

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2025 WSO2 LLC (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Processes a server response and extracts the result.
+#
+# + serverResponse - The response from the server, which may be a single JsonRpcMessage, a stream, or a transport error.
+# + return - Extracted ServerResult, ServerResponseError, or StreamError.
+isolated function processServerResponse(JsonRpcMessage|stream<JsonRpcMessage, StreamError?>|StreamableHttpTransportError? serverResponse)
+        returns ServerResult|ServerResponseError|StreamError {
+
+    // If response is a stream, extract the result from the stream.
```

Comment:
Most of the comments appear to state the obvious, as the code is already self-explanatory. Shall we remove them to improve readability? check other places too.